docs/extensions/custom-project-roles.md

```diff
@@ -0,0 +1,35 @@
+# Custom project roles
+
+The `Project` resource allows to specify a list of roles for every member (`.spec.members[*].roles`).
+There are a few standard roles defined by Gardener itself:
+
+* `owner` (describes the owner/main contact of the project)
+* `admin` (describes administrators with full read/write access to all resources concerning the project)
+* `viewer` (describes members with limited read access to some resources concerning the project)
+
+However, extension controllers running in the garden cluster may also create custom resources that project members might be able to CRUD.
+For this purpose Gardener also allows to specify custom roles.
+
+A custom role is prefixed with `custom:`, e.g.
+
+```yaml
+apiVersion: core.gardener.cloud/v1beta1
+kind: Project
+metadata:
+  name: dev
+spec:
+  members:
+  - apiGroup: rbac.authorization.k8s.io
+    kind: User
+    name: alice.doe@example.com
+    roles:
+    - "admin"
+    - "owner"
+    - "custom:foo"
+```
+
+The project controller will, for every custom role, create a `ClusterRole` with name `name: gardener.cloud:custom:project:<projectName>:<roleName>`, i.e., for above example: `name: gardener.cloud:custom:project:dev:foo`.
```

Comment:
Can you also add a sentence regarding the length restriction of custom role names?